packages/components/psammead-grid/src/index.jsx

```diff
@@ -0,0 +1,181 @@
+    gutterSize: GEL_GUTTER_ABOVE_600PX,
+    marginSize: GEL_MARGIN_ABOVE_400PX,
+  },
+  group5: {
+    min: GEL_GROUP_5_SCREEN_WIDTH_MIN,
+    max: null,
+    gutterSize: GEL_GUTTER_ABOVE_600PX,
+    marginSize: GEL_MARGIN_ABOVE_400PX,
+  },
+};
+
+const group4WrapperMaxWidth = `45.5rem`;
+// (6.75rem * 6) + 5*16px gutters = 728 = 45.5 rem
+
+const group5WrapperMaxWidth = `46.4rem`;
+// (2.95rem * 12) + 11*16px gutters = 742.4 = 46.4 rem
+
+const mediaQuery = ({ min, max, styles }) => {
+  if (min && max) {
+    return `
+      @media (min-width: ${min}) and (max-width: ${max}) {
+        ${styles}
+      }
+    `;
+  }
+  if (min) {
+    return `
+      @media (min-width: ${min}) {
+        ${styles}
+      }
+    `;
+  }
+  if (max) {
+    return `
+      @media (max-width: ${max}) {
+        ${styles}
+      }
+    `;
+  }
+  return '';
+};
+
+const gelMaxWidths = css`
+  margin: 0 auto;
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
+    max-width: ${group4WrapperMaxWidth};
+  }
+  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
+    max-width: ${group5WrapperMaxWidth};
+  }
+`;
+
+const group4MaxWidth = css`
+  margin: 0 auto;
+  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
+    max-width: ${group4WrapperMaxWidth};
+  }
+`;
+
+const gridMediaQueries = ({
+  columns,
+  startOffset,
+  enableGelGutters,
+  enableGelMargins,
+}) => {
+  const selectedGroups = Object.keys(columns);
+  return selectedGroups.map(group =>
+    mediaQuery({
+      min: groups[group].min,
+      max: groups[group].max,
+      styles: `
+        width: initial;
+        grid-template-columns: repeat(${columns[group]}, 1fr);
+        grid-column-end: span ${columns[group]};
+      ${enableGelGutters ? `grid-column-gap: ${groups[group].gutterSize};` : ``}
+      ${enableGelMargins ? `padding: 0 ${groups[group].marginSize};` : ``}
+      ${startOffset ? `grid-column-start: ${startOffset[group]};` : ``}`,
+    }),
+  );
+};
+
+const gridFallbacks = css`
+  ${({ wrapper, columns, parentColumns }) => {
+    if (wrapper || !parentColumns) {
+      return `position: relative;`;
+    }
+    const selectedGroups = Object.keys(columns);
+    return selectedGroups.map(
+      group =>
+        `
+      ${mediaQuery({
+        min: groups[group].min,
+        max: groups[group].max,
+        styles: `
+          display: inline-block;
+          width: ${(100 * columns[group]) / parentColumns[group]}%;
+          vertical-align: top;
+        `,
+      })}
+    `,
+    );
+  }}
+`;
+
+const GridComponent = styled.div`
+  ${({ enableGelMaxWidths }) => enableGelMaxWidths && gelMaxWidths}
+  ${({ enableGroupFourMaxWidth }) => enableGroupFourMaxWidth && group4MaxWidth}
+  ${gridFallbacks}
+  @supports (display: grid) {
+    ${gridMediaQueries}
+    ${({ wrapper }) =>
+      wrapper ? `display: grid; position: initial;` : `display: block;`}
+  }
+`;
+
+const Grid = ({ children, ...otherProps }) => {
```

Comment:
Need logic to ensure that a child cant have more columns than its parent

Reply:
This logic will presumably need to take into account the `startOffset` prop of the child (or what it gets renamed to).

Reply:
This is handled automatically by the current setup - a child cannot span more grid columns than the number of columns defined in its parent.